lib/src/conflicts.rs

```diff
@@ -427,9 +455,10 @@ pub fn update_conflict_from_content(
         return Ok(Some(conflict_id.clone()));
     }
 
-    let mut removed_content = vec![vec![]; conflict.removes.len()];
-    let mut added_content = vec![vec![]; conflict.adds.len()];
-    if let Some(hunks) = parse_conflict(content, conflict.removes.len(), conflict.adds.len()) {
+    let mut conflict = Conflict::from_backend_conflict(&conflict);
+    let mut removed_content = vec![vec![]; conflict.removes().len()];
+    let mut added_content = vec![vec![]; conflict.adds().len()];
+    if let Some(hunks) = parse_conflict(content, conflict.removes().len(), conflict.adds().len()) {
```

Comment:
It seems like our MSRV is one version short of allowing us to use let-else. Still, I think it's better to reduce the nesting follows:

```rust
// TODO: Change to let-else once our MSRV is above 1.65
let hunks = if let Some(h) = parse_conflict(content, conflict.removes().len(), conflict.adds().len()) {
  h
} else { 
  // Either there are no conflict markers of they don't have the expected arity
  return Ok(None)
}
```

I also think a comment like the one I put above would be nice to add. I'm not 100% sure this fixes all the possible similar bugs. (I don't think it's worth hunting for them, though, since changing to tree-level conflicts should make them less relevant).

See also my next comment about another TODO to maybe add here.

Reply:
Also, it is neat that this happens to fix a bug. Nice job noticing it.